### PR TITLE
fix(FEC-9513): endless spinner when auto play is failed

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

* Triggering `AD_AUTOPLAY_FAILED` once the play promise failed
* Add `BumperBreakType` enum
* Get the error code for `playOnMainVideoTag: true`

Depends on https://github.com/kaltura/playkit-js/pull/424

Solves [FEC-9513](https://kaltura.atlassian.net/browse/FEC-9513)

### CheckLists

* [x] changes have been done against master branch, and PR does not conflict
* [x] new unit / functional tests have been added (whenever applicable)
* [x] test are passing in local environment
* [x] Travis tests are passing (or test results are not worse than on master branch :))
* [ ] Docs have been updated

### CODE REVIEW

File: src/bumper.js [hunk 1]
```diff
@@ -20,10 +20,18 @@ import {BumperAdsController} from './bumper-ads-controller';
 import {BumperEngineDecorator} from './bumper-engine-decorator';
 import './assets/style.css';
 
+/**
+ * @enum {Object.<string, number>}}
+ */
+const BumperType: {[string]: number} = {
+  PREROLL: 0,
+  POSTROLL: -1
+};
```
Comment (RoyBregman) [inline src/bumper.js]: If the values represent the ad break position then shouldn't we call it BumperTypePosition or something?

File: test/src/bumper.spec.js [hunk 2]
```diff
@@ -953,6 +1001,29 @@ describe('Bumper', () => {
       player.play();
     });
 
+    it('Should fire AD_PLAY_FAILED when the play promise failed', done => {
```
Comment (RoyBregman) [inline test/src/bumper.spec.js]: whats the difference between this and the one in line 572?

File: src/bumper.js (hunk 1 above)
Comment (yairans) [reply src/bumper.js]: renamed to `BumperBreakType`. compare with https://github.com/kaltura/playkit-js/blob/fc25d940980eebfb9e2fb65611429e16d725ebea/src/ads/ad-break-type.js#L2

File: test/src/bumper.spec.js (hunk 2 above)
Comment (yairans) [reply test/src/bumper.spec.js]: we have to 2 modes for bumper - with dedicated video element and with the playback video element. these 2 tests are in a different mode.

File: src/bumper.js (hunk 1 above)
Comment (RoyBregman) [reply src/bumper.js]: so i guess both should be changed if they represent positions... by ok... :)

File: test/src/bumper.spec.js (hunk 2 above)
Comment (RoyBregman) [reply test/src/bumper.spec.js]: ok then

**Pull request merged**